integration-tests/src/main/scala/za/co/absa/spline/test/LineageWalker.scala

```diff
@@ -38,14 +38,9 @@ class LineageWalker(
 
   private def dependsOnRec(ref: AttrOrExprRef, id: String): Boolean = ref match {
     case AttrRef(attrIfd) =>
-      if (attrIfd == id) true
-      else dependsOnRec(attrMap(attrIfd).childRefs, id)
+      attrIfd == id || dependsOnRec(attrMap(attrIfd).childRefs, id)
     case ExprRef(exprId) =>
-      if (exprId == id) true
-      else {
-        if (litMap.contains("exprId")) false
-        else dependsOnRec(funMap(exprId).childRefs, id)
-      }
```

Comment:
It's shorter, but for me, the new version is harder to understand.

Reply:
What about introducing an intermediate val? For example?
```scala
lazy val attrChildIds = attrMap(attrId).childRefs
lazy val anyChildRefsDepends = dependsOnRec(attrChildRefs, id)
attrIfd == id || anyChildRefsDepends
```

The main motivation was to comply with the Sonar [rule](https://sonarcloud.io/project/issues?resolved=false&severities=MINOR&id=AbsaOSS_spline-spark-agent&open=AX7fDAUrz4PBF_ZPoJ7s&tab=why).

Subjectively, I understand why this is usually recommended approach. First, it's shorter, but also it is more plural for the reader. There is less nesting and branching involved for the reader. You have entire resulted condition written in one sentence that IMO more looks more natural, e.g. `condition = this or that and not something_else`. But I agree that a lot depends on naming. So I wouldn't hesitate choosing better names for functions, or introducing intermediate variables.

On a side note, I just noticed that `attrIfd` is probably a typo. Shouldn't it be `attrId`? Also `dependsOnRef` instead of `dependsOnRec`?

Reply:
For simple cases, It may be ok, but here the second condition is more complex. The nesting and branching is still there, it just now must happen inside the reader's head instead of in the code. The reader must consider short-circuiting, negate one expression and operator precedence, and then build a truth table to make sense of it. The Sonar just doesn't consider any of that.

Shorter is not better, I think we should aim to minimize cognitive load.

Why do you mean by `plural`?

`attrIfd` yes typo
`dependsOnRec` no it means recursive - just private name

Reply:
Let me refactor it a bit and I'll propose another variant.